### ISSUE TITLE
concept mappings need to be taken with care

### ISSUE THREAD

**Issue by LinguList** [martino-vic/gerstnerhungarian#3]
The automated procedure surely has its advantages, but it is dangerous to use it without checking the data, and in this case, this becomes very clear. Since the senses offered by the author are so numerous, finding a hit is trivial, but this hit may often be misleading. So I would suggest to raise the bars and think of a future workflow by which we can avoid that you will end up comparing apples with pears, due to problematic mappings, since Concepticon was designed to make concept comparisons more exact in linguistics.

**Comment by LinguList** [comment]
An idea we had for some time, but which we never finished, was to find a strategy how to extract a wordlist from a dictionary, which would be in part manually curated.

For your purpose, we could design a workflow by which you have a master-conceptlist, like the IDS or similar, where you have WOLD borrowability mappings, and then you run this against the data in your dictionary, which will yield fewer mappings, but you can refine them manually.

**Comment by LinguList** [comment]
This could again be turned into a blogpost, or one could consider a common publication where we present this as a new way to deal with dictionaries (then exemaplary, more abstract).

**Comment by martino-vic** [comment]
This is of course a bit hard to understand for me at a first glance, but I think I get the main idea. I always enjoy writing blog posts, and I would be excited about a common publication as well, thank you for the suggestion! I would be willing to put some energy into this matter to understand it better.

**Comment by LinguList** [comment]
So to try putting it in other words: the mappings we permit to Concepticon are always strict, never vague. If a gloss is "green kind of plant", we prefer not mapping it, as we do not know what concept this denotes. 

But a dictionary is not a concept list, but made of senses, as you find in your file here, like "axe, instrument, sharp argument" (I am making this one up, but you see that the senses are really different, but that you would probably be able to say that the word can express the concept "axe").

So to turn a dictionary into a word list, we need to have a basic number of concepts and see which entries with senses express these concepts. For contemporary languages, this is easier, but one can also do it more or less for Ancient languages. 

The task is then, you have the concept list, your questionnaire, to find those words in your dictionary expressing those concepts, and you can use the dictionary senses for the mapping.

Right now, you interpret the multiple senses as a concept elicitation gloss, which they aren't. 

So we would turn it all around, and use a master list of concepts, and then check sense by sense, if this concept is expressed by the sense description. We then assign the word form to the concept, and refine them all later on.

In this way, you have a wordlist, not a dictionary.

The CLDF structure will change here, as you'll have two files: a dictionary file and a wordlist file.

**Comment by LinguList** [comment]
Our first step would be to make some master concept list, which should be a bit limited, some 1000 or so entries, I suggest.

**Comment by martino-vic** [comment]
Ah, now I get it, but that task has been done already: http://northeuralex.org/languages/hun

**Comment by LinguList** [comment]
You mean making the master conceptlist? Is that the reference list you want to use for your work?

**Comment by LinguList** [comment]
Yes, there is a misunderstanding. I will try to explain later.

**Comment by LinguList** [comment]
It is not about making a different mapping, but a different kind of dataset. What you have is currently problematic, as it is a dictionary that you mask as a wordlist. A wordlist starts from concepts and provides words that express these. So a wordlist can have identifcal forms expressing different concepts. A dictionary starts from forms and provides descriptions of their meaning. Meanings can consist of more than one concept. For dictionaries and wordlists, we have CLDF datasets, but you use the wordlist dataset. So what you should first do for a dictionary is using a dictionary structure, and then we can discuss how to add a manually curated wordlist derived from the dictionary on top of that. 

So what we should do first is following the cldf-code shown above to make a dictionary out of the data, with senses, etc. Then, we make a script that can hopefully be reused to derive a wordlist from the dictionary, potentially one that reflects the 1005 concepts (or 1004?) in NorthEuralex.

**Comment by martino-vic** [comment]
Okay that makes sense, now I know exactly what I need to do, thank you for the detailed explanation :)

**Comment by martino-vic** [comment]
Okay, so I've looked into how to create a dictionary but it seems quite different from wordlists. If I see it right, I'll have to use pydictionaria in the conversion-script and provide the raw data as a .sfm file, right?

**Comment by LinguList** [comment]
Yes and no. The underlying data can be any format. But maybe let me start by setting this up for the streitberggothic dictionary? I'd ask colleagues to review, whom I will inform about this, as we wanted to do this for a long time and were experimenting with other dictionaries.

**Comment by LinguList** [comment]
I cannot promise when this will be done, but I'll put it onto my todo-list, and I have already contacted our colleagues.

**Comment by martino-vic** [comment]
That is great to hear, thank you very much, there is no rush :)

**Comment by martino-vic** [comment]
I've applied the new worflow from [streitberggothic](https://github.com/martino-vic/streitberggothic/pull/11) and pushed the repo. Still need to manually review wordlist.tsv tho.

**Comment by martino-vic** [comment]
the automatic refinement wasn't good so I went through it manually and pushed the repo
